core/src/validator.rs

```diff
@@ -1409,6 +1439,19 @@ impl Validator {
         let cluster_slots =
             Arc::new(crate::cluster_slots_service::cluster_slots::ClusterSlots::default());
 
+        // bank_forks could be write-locked temporarily here, if unified scheduler is enabled for
+        // block production. That's because ReplayStage started inside Tvu could immediately try to
+        // insert a tpu bank into bank_forks, like with local development clusters consisting of a
+        // single staked node. Such insertion could be blocked with the lock held by unified
+        // scheduler until it's fully setup by the banking stage. This is intentional because
+        // completion of insertion needs to strictly correspond with the initiation of producing
+        // blocks in unified scheduler. Because Tpu setup follows Tvu setup, there's a corner case
+        // where the banking stage hasn't yet called register_banking_stage() to finish the unified
+        // scheduler setup.
+        // This means any setup which accesses bank_forks must be done before
+        // or after the short duration of Tvu and Tpu setups to avoid deadlocks. So,
+        // RootBankCache needs to be created here in advance as being one of such setups.
```

Comment:
this

Reply:
See the above comment on the lock order - feel it may be worth investing in a better way to deliver working banks to tpu to avoid these weird edge cases where we need to be very careful.

Reply:
it seems that that wall of text couldn't help and i crossed your code-smell threshold nevertheless.. ;)

how about this? 110c1bcdb7d2b6f92208879f793d2d7290fbc2f8 (note: ci won't pass)

if it's okay for you that unified scheduler code breaks out of the BankingStage code organization more aggressively like that, i can remove this weird edge case: 9839ab82a91df4276f6ba0293b0c00b608fb770e

> feel it may be worth investing in a better way to deliver working banks to tpu to avoid these weird edge cases where we need to be very careful.

fyi, this is completely different from the `update_bank_forks_and_poh_recorder_for_new_tpu_bank` thing (tpu delivery as you said), which i'll plan to address later with a different commit.